Cognite.Simulator.Utils/ModelLibraryBase.cs

```diff
@@ -297,6 +304,8 @@ private async Task<T> GetOrAddModelRevisionImpl(string modelRevisionExternalId,
                 await ExtractModelInformationAndPersist(modelState, token).ConfigureAwait(false);
             }
 
+            modelState.DownloadAttempts = 0;
```

Comment:
this should be only reset after the model is downloaded, otherwise we get an endless retry loop
I think it's best to put it inside the `DownloadAllModelFilesAsync`

before the end of the function we could do:
```
if (allFilesDownloaded) {
    modelState.DownloadAttempts = 0;
}
```